src/foam/nanos/theme/customisation/ThemeCustomisation.js

```diff
@@ -0,0 +1,176 @@
+/**
+* @license
+* Copyright 2021 The FOAM Authors. All Rights Reserved.
+* http://www.apache.org/licenses/LICENSE-2.0
+*/
+
+/**
+ * TODO:
+ * previewViews
+ * rollbacks
+ */
+
+foam.CLASS({
+  package: 'foam.nanos.theme.customisation',
+  name: 'ThemeCustomisationBrowseController',
+  extends: 'foam.comics.v2.DAOBrowseControllerView',
+
+  properties: [
+    {
+      name: 'click',
+      expression: function() {
+        return function(obj, id) {
+          if ( ! this.stack ) return;
+          this.stack.push(foam.u2.stack.StackBlock.create({
+          view: {
+            class: 'foam.nanos.theme.customisation.ThemeCustomisationView',
+            themeName: id,
+          }, parent: this.__subContext__ }, this));
+        };
+      }
+    }
+  ]
+});
+
+foam.CLASS({
+  package: 'foam.nanos.theme.customisation',
+  name: 'ThemeCustomisationView',
+  extends: 'foam.u2.View',
+
+  imports: ['subject', 'spThemeDAO', 'theme', 'ctrl'],
+
+  exports: ['controllerMode', 'data'],
+
+  requires: [
+    'foam.u2.Tab',
+    'foam.u2.Tabs',
+    'foam.nanos.theme.customisation.ThemeComponentView',
+    'foam.nanos.theme.customisation.ThemeFacade',
+    'foam.u2.stack.BreadcrumbView',
+    'foam.u2.layout.Rows',
+    'foam.nanos.theme.Theme',
+    'foam.layout.SectionAxiom',
+    'foam.layout.Section',
+    'foam.u2.borders.CardBorder',
+    'foam.u2.detail.SectionedDetailPropertyView'
+  ],
+
+  css: `
+    ^ {
+      padding: 32px;
+    }
+    ^container > * + * {
+      margin-top: 32px;
+    }
+    ^button {
+      align-self: flex-end;
+    }
+    ^sectionWrapper {
+      display: flex;
+      flex-direction: column;
+    }
+    ^sectionWrapper > * + * {
+      margin-top: 8px;
+    }
+  `,
+
+  messages: [
+    { name: 'THEME_MSG', message: 'Theme' },
+    { name: 'UPDATED_MSG', message: 'updated' },
+    { name: 'ERROR_MSG', message: 'Error' },
+  ],
+
+  properties: [
+    {
+      name: 'data',
+      visibility: 'HIDDEN',
+      factory: function() {
+        this.ThemeFacade.create();
+      }
+    },
+    {
+      class: 'String',
+      name: 'themeName'
+    },
+    {
+      class: 'Map',
+      name: 'propViews_',
+      visibility: 'HIDDEN',
+      value: {}
```

Comment:
Remove this line. It would just cause different instances to share the same value, and isn't needed because Map already does:
    [ 'factory', function() { return {} } ],